diff -Nur tvision.orig/include/tv/inputln.h tvision.chng/include/tv/inputln.h
--- tvision.orig/include/tv/inputln.h	2003-06-02 17:03:30.000000000 +0200
+++ tvision.chng/include/tv/inputln.h	2004-06-15 12:29:08.016799864 +0200
@@ -245,14 +245,13 @@
 {
 public:
  TInput1Line(int x, int y, int max) :
-   TInputLine(TRect(x,y,x+max+2,y+1), max)) {};
+   TInputLine(TRect(x,y,x+max+2,y+1), max) {};
 };
 
 class TInput1LineU16 : public TInputLineU16
 {
 public:
  TInput1LineU16(int x, int y, int max) :
-   TInputLineU16(TRect(x,y,x+max+2,y+1), max)) {};
+   TInputLineU16(TRect(x,y,x+max+2,y+1), max) {};
 }
 #endif // Uses_TInput1Line
-
