--- tvision/config.pl.old	2011-05-18 12:17:36.000000000 +0200
+++ tvision/config.pl	2011-05-18 12:19:48.000000000 +0200
@@ -255,7 +255,7 @@ if ($OS eq 'UNIX')
    $aux='';
    $aux.=" -L".$conf{'X11LibPath'}." $libs" if @conf{'HAVE_X11'} eq 'yes';
    $aux.=' -lgpm' if @conf{'HAVE_GPM'} eq 'yes';
-   $aux.=(($OSf eq 'QNXRtP') ? ' -lncursesS' : ' -lncurses') unless $conf{'ncurses'} eq 'no';
+   $aux.=(($OSf eq 'QNXRtP') ? ' -lncursesS' : ' -lncursesw') unless $conf{'ncurses'} eq 'no';
    $aux.=" $stdcxx -lm -lc";
    $aux.=' -lpthread' if $conf{'HAVE_LINUX_PTHREAD'} eq 'yes';
    $aux.=' '.$AllegroLibs if ($conf{'HAVE_ALLEGRO'} eq 'yes');
@@ -870,7 +870,7 @@ sub LookForNCurses
  # Assume it is -lncurses
  if ($OSf ne 'QNXRtP')
  {
-  $conf{'NameCurses'}='ncurses'
+  $conf{'NameCurses'}='ncursesw'
  }
  else
  {
