--- tvision/include/tv/linux/screen.h.orig	2003-05-21 14:27:00.000000000 +0200
+++ tvision/include/tv/linux/screen.h	2003-10-09 20:03:52.179916896 +0200
@@ -224,13 +224,5 @@
 // some problems when compiling for Alpha (__alpha__).
 //   Also make it only for Linux until I know how to do it for FreeBSD.
 
-#if defined(TVCPU_x86)
- // Needed for ioperm, used only by i386.
- // I also noted that glibc 2.1.3 for Alpha, SPARC and PPC doesn't have
- // this header
- #include <sys/perm.h>
- #define h386LowLevel
-#endif
-
 #endif // LINUXSCREEN_HEADER_INCLUDED
 
